### DISCUSSION TITLE
[BGFIX] Debian 11 Error while running 'make'

### DISCUSSION THREAD

**Discussion by OposDev** [Almamu/linux-wallpaperengine#119] [Q&A]
**Console output**
`-- Checking for module 'xrandr'
--   No package 'xrandr' found
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Xrandr (missing: Xrandr_LIBRARY Xrandr_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  CMakeModules/FindXrandr.cmake:64 (find_package_handle_standard_args)
  CMakeLists.txt:12 (find_package)`

**Screenshots**
![image](https://user-images.githubusercontent.com/58175621/176068552-ba1db9ad-f3ad-412d-962f-0ac38bea3bff.png)

I have Xrandr installed by using the X11-xserver-utils package (Debian does not have another xrandr package)

**Desktop (please complete the following information):**
 
- OS: Debian 11
 - Desktop Environment: XFCE4
 - Window Manager: i3WM

**Additional context**

Any additional information about your setup

**Comment by OposDev** [comment]
After 3 hours of searching for all the required packages, I managed to get this to finally compile, but when I ran `make` I got another error. 

![image](https://user-images.githubusercontent.com/58175621/176091332-9e40245d-0382-4ce9-bcd2-74c3977566ad.png)

I'm not sure if this has anything to do with the error above.
![image](https://user-images.githubusercontent.com/58175621/176092385-bc40c4e9-0a74-42a4-ba21-10795e519756.png)

**Comment by mihawk90** [comment]
I'd suggest you go through the list or requirements in the readme and install their respective development packages.

**Reply by OposDev** [reply]
I have all of the required packages, yet it still doesn't build properly. I added more info below.

**Comment by danoooob** [comment]
```
sudo apt install -y libsdl-mixer1.2-dev libxxf86vm-dev libfreeimage-dev libglfw3-dev libglm-dev
```
Some missing packages from your step

**Reply by OposDev** [reply]
I already had these packages installed, still getting the same error.

**Comment by OposDev** [comment]
I've installed the missing packages from above, but i'm still receiving the same error.

When running `cmake ..`, the terminal shows that all packages are found, except for this 
![image](https://user-images.githubusercontent.com/58175621/179282344-3b17cd97-1beb-46c2-9ec0-85ca9abf0222.png)

When I run `make`, the same error occurs 
![image](https://user-images.githubusercontent.com/58175621/179282413-c48bc42e-c6a5-4ec7-abd6-e89f6d30e5b6.png)

SDL_mixer.h is also installed:
![image](https://user-images.githubusercontent.com/58175621/179286643-42b2fc88-4531-4f5d-9c6e-8cfa171a369b.png)


Kind of confused as to why this issue was closed when it hasn't been solved?

**Reply by mihawk90** [reply]
> When running `cmake ..`, the terminal shows that all packages are found, except for this

That's irrelevant, it just tries to find various different implementations. The important part is the last line.

> When I run make, the same error occurs
> SDL_mixer.h is also installed:

I don't have a Debian install to test on, but what I can say is that it compiles fine on Fedora and Arch (which I think Almamu is on too).
Try deleting the entire build directory if you haven't already, that should at least get rid of any potential issues with the CMake or Make files.

> Kind of confused as to why this issue was closed when it hasn't been solved?

It's not closed, it was moved to a Discussion, because it's not an issue with the codebase and therefore not a bug.

**Reply by OposDev** [reply]
> That's irrelevant, it just tries to find various different implementations. The important part is the last line.

Ahh alright, that makes sense. 

> I don't have a Debian install to test on, but what I can say is that it compiles fine on Fedora and Arch (which I think Almamu is on too).
Try deleting the entire build directory if you haven't already, that should at least get rid of any potential issues with the CMake or Make files.

I've deleted the build directory multiple times, it continually fails to find SDL_mixer.h. I've also attempted to move the directory to the home directory in hopes of changing something, same error. 

> It's not closed, it was moved to a Discussion, because it's not an issue with the codebase and therefore not a bug.

Ahh okey, is there anything else that I can do to attempt fixing this? Maybe input something into the makefile to help it locate the SDL_mixer.h file? 

**Edit:** I looked through the CMakeCache file and it's able to find the SDL directory containing the SDL_mixer.h file, so I'm not sure why ``make`` isn't working :/

**Comment by OposDev** [comment]
UPDATE: I got it to build! I looked around in the CMakeCache and found that it was searching in `/usr/include/` but couldn't find the SDL folder that APT put the SDL headerfiles in, so i moved all of the header files into the include directory, and it completed building :D 

![image](https://user-images.githubusercontent.com/58175621/179358666-d040ec4e-15f8-4f5c-b369-2bdcb1b6f30d.png)

I'm receiving a new error now, `Detected X Error, X Error of failed request: BadWindow (Invalid window parameter`

Extra info: 
![image](https://user-images.githubusercontent.com/58175621/179360220-5a57a474-396c-4c6e-b0b4-1adad0d5c242.png)
![image](https://user-images.githubusercontent.com/58175621/179360503-ec68b069-3711-42dd-b760-43e9a72debbd.png)

**Reply by Almamu** [reply]
That's weird, SDL_mixer should be found without moving them to a different folder (cmake should be taking care of that)... maybe there's something wrong with the header finding of cmake?

About the X11 error, please make sure you're running the latest version and that there's no DE running, just i3, NVIDIA drivers might also be a bit problematic, does it run properly if you run it in window mode (without the --screen-root options?)

**Reply by OposDev** [reply]
I'm not entirely sure about the cmake file. It looked fine to me, but ye, failed to find the SDL header file directory. 

For the X11 error, am running the latest version and have confirmed that there is no DE running. 
The application runs properly in window mode, but fails to assign itself to the desktop background. 

Would the Nvidia drivers become an issue? Am currently running a combination of Xwinwrap, Xdotools and MPV to run animated wallpapers and it seems to work fine with the proprietary nvidia drivers :D

**Comment by OposDev** [comment]
Update 2: I was testing different types of wallpapers (mp4 and shaders) and have noticed a drop in resolution quality. Is this expected? 

![image](https://user-images.githubusercontent.com/58175621/186534713-d17240d8-0cf0-4d23-a743-487844e99139.png)

In this wallpaper, the pixelation is highly visible, 

![image](https://user-images.githubusercontent.com/58175621/186534820-2fe5e90e-5bf8-4b34-8fcd-f20deda10355.png)

In this wallpaper, the pixelation is still there, but not as noticeable.

Have not managed to get the software to write wallpapers to the background yet..
